handlers/pools.go

```diff
@@ -0,0 +1,268 @@
+package handlers
+
+import (
+	// "eth2-exporter/db"
+
+	"encoding/json"
+	"eth2-exporter/db"
+	"eth2-exporter/services"
+	types "eth2-exporter/types"
+	"eth2-exporter/utils"
+	"fmt"
+	"html/template"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+	// "strings"
+)
+
+var poolsServicesTemplate = template.Must(template.New("poolsServices").Funcs(utils.GetTemplateFuncs()).ParseFiles(
+	"templates/layout.html",
+	"templates/poolsServices.html",
+	"templates/bannerPoolsServices.html",
+	"templates/index/depositDistribution.html"))
+
+type pools struct {
+	Address  string  `db:"address" json:"address"`
+	Name     string  `db:"name" json:"name"`
+	Deposit  int64   `db:"deposit" json:"deposit"`
+	Category *string `db:"category" json:"category"`
+}
+
+type poolInfo struct {
+	Status         string `db:"status" json:"status"`
+	ValidatorIndex uint64 `db:"validatorindex" json:"validatorindex"`
+	Balance31d     uint64 `db:"balance31d" json:"balance31d"`
+}
+
+type poolStatsData struct {
+	PoolInfo []poolInfo
+	Address  string
+}
+
+type chart struct {
+	DepositDistribution types.ChartsPageDataChart
+	StakedEther         string
+	PoolInfo            []respData
+	EthSupply           interface{}
+	LastUpdate          int64
+}
+
+type respData struct {
+	Address    string                   `json:"address"`
+	Name       string                   `json:"name"`
+	Deposit    int64                    `json:"deposit"`
+	Category   *string                  `json:"category"`
+	PoolInfo   []poolInfo               `json:"poolInfo"`
+	PoolIncome *types.ValidatorEarnings `json:"poolIncome"`
+}
+
+var poolInfoTemp []respData
+var poolInfoTempTime time.Time
+var ethSupply interface{}
+var updateMux = &sync.RWMutex{}
+var firstReq = true
+
+func Pools(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/html")
+
+	data := InitPageData(w, r, "services", "/pools", "Stacking Pools Services Overview")
+
+	chartData, err := services.ChartHandlers["deposits_distribution"].DataFunc()
+	if err != nil {
+		logger.Errorf("error executing template for %v route: %v", r.URL.String(), err)
+		http.Error(w, "Internal server error", 503)
+		return
+	}
+
+	var pieChart chart
+
+	indexStats := services.LatestIndexPageData()
+
+	pieChart.DepositDistribution.Data = chartData
+	pieChart.DepositDistribution.Height = 500
+	pieChart.DepositDistribution.Path = "deposits_distribution"
+	pieChart.StakedEther = indexStats.StakedEther
+
+	updateMux.Lock()
```

Comment:
It would make more sense to have this in a service instead of in the handler. We should be able to call handlers concurrently.

Reply:
this is one of the thing that concerns me too, I wanted to see if it becomes a bottleneck, but you are right maybe I should move it to services